kickoff/jsonexport.py

```diff
@@ -0,0 +1,244 @@
+from kickoff import app
+
+from flask import jsonify, render_template
+
+from kickoff.model import getReleases
+from kickoff.firefoxversions import firefoxversions
+
+from kickoff.thunderbirddetails import primary_builds as tb_primary_builds, beta_builds as tb_beta_builds
+from kickoff.thunderbirdversions import thunderbirdversions
+
+from kickoff.mobileversions import mobileversions
+
+import config
+
+
+def getFilteredReleases(product, categories, ESR_NEXT=False, lastRelease=None, withL10N=False):
+    version = []
+    # we don't export esr in the version name
+    if "major" in categories:
+        version.append("([0-9]+\.[0-9]+|14\.0\.1)$")
+    if "stability" in categories:
+        version.append("([0-9]+\.[0-9]+\.[0-9]+|[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)")
+    if "dev" in categories:
+        version.append("[0-9]+\.[0-9](b|rc|build|plugin)[0-9]+$")
+    if "esr" in categories:
+        if ESR_NEXT:
+            # Ugly hack to manage the next ESR (when we have two overlapping esr)
+            version.append("(" + config.ESR_NEXT + "\.[0-9]+\.[0-9]+esr|" + config.ESR_NEXT + "\.[0-9]+\.[0-9]+\.[0-9]+esr)")
+        else:
+            version.append("(" + config.CURRENT_ESR + "\.[0-9]+\.[0-9]+esr|" + config.CURRENT_ESR + "\.[0-9]+\.[0-9]+\.[0-9]+esr)")
+
+    releases = getReleases(ready=True, productFilter=product, versionFilterCategory=version, lastRelease=lastRelease)
+    results = []
+    for r in releases:
+        version = r.version.replace("esr", "")
+        if r._shippedAt:
+            shippedAt = r._shippedAt.strftime("%Y-%m-%d")
+        else:
+            # No shipped date, we are not interested by this release, skip
+            continue
+
+        if withL10N:
+            # We need the list of locales
+            results.append([version, shippedAt, r.l10nChangesets])
+        else:
+            results.append([version, shippedAt])
+
+    return results
+
+# Firefox JSON
+
+@app.route('/json/firefox_history_major_releases.json', methods=['GET'])
+def firefox_history_major_releases_json():
+    # Match X.Y and 14.0.1 (special case)
+    values = getFilteredReleases("firefox", "major")
+    return jsonify(values)
+
+
+@app.route('/json/firefox_history_stability_releases.json', methods=['GET'])
+def firefox_history_stability_releases_json():
+    # Match X.Y.Z (including esr) + W.X.Y.Z (example 1.5.0.8)
+    values = getFilteredReleases("firefox", "stability")
+    return jsonify(values)
+
+
+@app.route('/json/firefox_history_development_releases.json', methods=['GET'])
+def firefox_history_development_releases_json():
+    # Match 23.b2, 1.0rc2, 3.6.3plugin1 or 3.6.4build7
+    values = getFilteredReleases("firefox", "dev")
+    return jsonify(values)
+
+
+@app.route('/json/firefox_versions.json', methods=['GET'])
+def firefox_versions_json():
+    # Stable
+    lastStable = getFilteredReleases("firefox", ["major", "stability"], lastRelease=True)
+    firefoxversions["LATEST_FIREFOX_VERSION"] = lastStable[0][0]
```

Comment:
Am I reading the new diff correctly that you took a new approach and eliminate the need for a file like firefoxversions.py altogether? That's totally fine, just double checking!